Provide an Escrow-as-Smart Contract use case

jhaaaa opened the pull request:
- Provides a use case that describes how to use a conditionally-held escrow as a smart contract

jhaaaa committed:
smart contract use case

jhaaaa committed:
wip - need to test flow

jhaaaa committed:
still working

jhaaaa committed:
still working

jhaaaa committed:
first draft

jhaaaa commented:
@mDuo13 , @warpaul , & @garay-daniel - Sorry for the delay in getting this use case out to review with you. When you have a moment, please take a look and feel free to comment. As I mention in a TODO comment in the markdown, I'm using a far-out scenario right now. (What, you don't know party planners who want to generate condition and fulfillment values? 😂) But I needed to use a dead simple scenario to help me focus on and work through the logistics. I'm open to changing up the scenario! Thank you.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+A smart contract can help you exchange anything of value in a transparent, traceable, tamper-resistant, and irreversible way.
+
+The benefit of encoding a smart contract into a blockchain is that it enables the contract to be securely performed without traditional third-parties, like financial or legal institutions, which can result in lower costs. Instead, the contract is supervised by the distributed, decentralized network of computers that run the blockchain.
```
I think "which can result in lower costs" is a misplaced modifier here. Also, <sup>[citation needed]</sup>, so maybe just leave it off.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+A smart contract is a blockchain-based program that encodes the conditions and fulfillment of an agreement between two or more parties and automatically fulfills the terms of the agreement once conditions are met.
+
+A smart contract can help you exchange anything of value in a transparent, traceable, tamper-resistant, and irreversible way.
```
Maybe combine with the previous paragraph?

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+Let's use this scenario to help illustrate this use case: A party planner uses smart contracts to manage payments from party hosts to party vendors. Specifically, the party planner wants to use a smart contract to have the party host pay the party band 2000 XRP once they are done with their set. ***TODO: this is a far-out example -- but I had to use something really tangible to be able to make my ...
+
+In this use case, the party host is the sender of the escrow, the party band is the receiver of the escrow, and the party planner is playing the role of an _oracle_. In the context of smart contracts, an oracle is a neutral third-party agent that can verify real-world events to either fulfill or invalidate a smart contract. This use case uses a human oracle for illustrative purposes, but in real- ...
```
I wish we had an example that used an actual software oracle. I'll see if I can think of a scenario that's a little more "true to life". For now, this is OK.

mDuo13 replied:
Suggestion:
```
< a software application would more likely play the role of the oracle to automate the process.
---
> a software application would more likely play the role of the oracle.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+In this use case, the party host is the sender of the escrow, the party band is the receiver of the escrow, and the party planner is playing the role of an _oracle_. In the context of smart contracts, an oracle is a neutral third-party agent that can verify real-world events to either fulfill or invalidate a smart contract. This use case uses a human oracle for illustrative purposes, but in real- ...
+
+Using an XRP Ledger escrow to provide this smart contract is a great arrangement because the party planner, as the third-party oracle, never "holds" the funds as one might in a traditional escrow arrangement, and can't possibly take the funds for themselves.
```
:+1:

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+- An XRP Ledger [account](accounts.html#creating-accounts) that holds enough XRP to pay for escrow and any fees incurred.
+
+- Access to a [`rippled` server](install-rippled.html) that they can use to create the escrow.
```
Technically they only need a [secure signing environment](https://github.com/ripple/ripple-dev-portal/pull/506), I think. Which should include having a network connection to a `rippled` server (any server) they can submit their signed transaction to.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+- The ability to generate a condition and a fulfillment.
+
+- Access to a [`rippled` server](install-rippled.html) that they can use to submit the fulfillment value and finish the escrow.
```
Having a rippled server and the ability to finish the escrow should be under the band's requirements. (You need an XRP Ledger account to send any kind of transaction, after all.) The payee (the band) can be the one to actually construct and submit the transaction to get paid, after the oracle tells them what fulfillment to use. 

The party planner (the oracle) needs:

- To be able to keep a secret (the fulfillment) until the time is right.
- A way to communicate the fulfillment publicly or at least to the payee (the band) when the time is right.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+To create the escrow as a smart contract, the participants must first define the terms of the contract. In this scenario, the participants need to agree on the following details.
+
+- Should the escrow be fulfilled after a specific time has passed or after a cryptographic condition has been fulfilled?
```
Feels a little odd to ask this here after this whole use case has been building up using the crypto-condition.

jhaaaa replied:
Thank you for this! Sure thing. I included it because it points out that you have these options to pick from when you are creating an escrow -- but you are right, the scenario has already picked the conditionally-held option!

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+- Should the escrow be fulfilled after a specific time has passed or after a cryptographic condition has been fulfilled?
+
+    - *[Conditionally-held escrow](send-a-conditionally-held-escrow.html)*: In this case, the participants agree to use a conditionally-held escrow. This is a good choice because the smart contract can be fulfilled only after the party planner (oracle) verifies that the party band completed their set and submits the cryptographic condition and fulfillment to finish the escrow and release the XRP.
```
As mentioned above, the oracle doesn't submit the transaction to finish the escrow. The oracle merely provides the information that makes it possible for the payee to finish the escrow.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+    - *[Conditionally-held escrow](send-a-conditionally-held-escrow.html)*: In this case, the participants agree to use a conditionally-held escrow. This is a good choice because the smart contract can be fulfilled only after the party planner (oracle) verifies that the party band completed their set and submits the cryptographic condition and fulfillment to finish the escrow and release the XRP.
+
+    To handle a no-show scenario, the participants configure the escrow to be eligible for cancellation after a specific time. Typically, this cancellation time should be after the time at which the party planner is expected to finish the escrow. The the cancellation time in this way helps prevent the party host from maliciously cancelling the escrow before the party planner can finish the escrow ...
```
This isn't quite right.

An XRP Ledger Escrow cannot be "maliciously" canceled. The escrow exists in three states (I just made up the names of these states for explanation. They're not used elsewhere):

- "Held." No one can finish or cancel it.
- "Ripe." Anyone can finish it. No one can cancel it.
- "Expired." Anyone can cancel it. No one can finish it.

A time-based escrow is "ripe" after the `FinishAfter` time has passed.

A conditional escrow is "ripe" after the oracle has published the fulfillment.

A time-and-conditional escrow is "ripe" after both have happened.

An escrow that passes its `CancelAfter` time is expired, regardless of what happened with the fulfillment. An escrow that doesn't have a `CancelAfter` time never expires. If you're using an oracle, you _definitely_ want to use `CancelAfter` so that you can get your money back if the oracle does not publish the fulfillment.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+    - *[Time-held escrow](send-a-time-held-escrow.html)*: A time-held escrow is not the best choice for this scenario because the party band could be a no-show. If the party band is scheduled to finish their set at 11 p.m., the participants could agree to allow the escrow to finish and release the XRP at 11 p.m. At 11 p.m., any participant, including the party band, can finish the escrow and rele ...
+
+    The participants could agree to allow the cancellation of the escrow before 11 p.m. to accommodate the no-show scenario. However, consider that the party band could show up and play their set, but the party host cancels the escrow before a participant can finish the escrow at 11 pm.
```
You can't use the cancellation on XRP Ledger escrows this way.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+- Should the escrow disallow fulfillment until a specific time?
+
+  The participants agree to disallow the escrow from being fulfilled until 11 p.m. the night of the party, which is when the party band is scheduled to complete their set. ***TODO: not necessary, but just an extra precaution? Or just unnecessary?***
```
Unnecessary. I don't think it provides any additional security, since whether the escrow can be finished still depends entirely on whether the oracle publishes the fulfillment before the expiration.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+  The participants agree to disallow the escrow from being fulfilled until 11 p.m. the night of the party, which is when the party band is scheduled to complete their set. ***TODO: not necessary, but just an extra precaution? Or just unnecessary?***
+
+- Should the participants be able to cancel the escrow after a specific time period if the condition is not fulfilled?
```
> Should the participants be able to cancel the escrow

Suggest changing this to "Should the escow expire" and the answer is "absolutely yes" because otherwise the only outcomes are:

- The party planner releases the condition: the band gets paid
- The party planner doesn't release the condition: the money stays locked in escrow forever

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+  The participants agree that the escrow can be cancelled after 12 noon the day after the party. This gives the party planner enough time to finish the escrow, if the party band fulfills their end of the contract. After cancellation, the locked XRP returns to the party host's account.
+
+  This timing also prevents the party host from maliciously cancelling the escrow before the party planner can finish the fulfilled contract.
```
As mentioned, remove this line; that's not how cancellation works.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+- Should the participants be able to cancel the escrow after a specific time period if the condition is not fulfilled?
+
+  The participants agree that the escrow can be cancelled after 12 noon the day after the party. This gives the party planner enough time to finish the escrow, if the party band fulfills their end of the contract. After cancellation, the locked XRP returns to the party host's account.
```
Suggest changing to:

> This gives the band enough time to finish the escrow, after the party planner verifies that they fulfilled their end of the contract and publishes the cryptographic fulfillment.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+  The participants agree that the escrow should lock up and potentially pay 2000 XRP, which is the party band's fee.
+
+- Which XRP Ledger account should the escrow lock up and potentially pay XRP out of?
```
This wording implies that the escrow "locks up" an account. It only locks up the specified amount of XRP, not the whole account.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+  The participants agree that the escrow should lock up and potentially pay XRP out of the party host's XRP Ledger account.
+
+- Which XRP Ledger account should the escrow lock up and potentially pay XRP into?
```
I think this sentence reads more clearly if you remove "lock up and" from it:

> Which XRP Ledger account should the escrow potentially pay XRP to?

(Also, I like "pay XRP to" rather than "pay XRP into" here, but that's not that important.)

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+The party planner generates the condition and fulfillment values and provides the condition value to the party host, who creates the escrow.
+
+The party planner must keep the fulfillment value a secret. They will use the condition and fulfillment values to finish the escrow.
```
> They will use the condition and fulfillment values to finish the escrow.

Anyone can use these, but most likely the band will since they're the ones who're motivated to get paid.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+Because the participants want the escrow to be eligible for cancellation after 12 noon the day after the party, the party host (sender) must calculate a `CancelAfter` value to include in the escrow definition.
+
+Because the participants want to disallow the escrow from being finished until after 11:00 pm the night of the party (the time when the party band is scheduled to finish its set), the party host must calculate a `FinishAfter` value to include in the escrow definition.
```
Since I don't think specifying a `FinishAfter` alongside a condition is useful, scratch this.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+The party host (sender) waits for validation of the ledger that contains the escrow creation transaction and then confirms that the escrow was created.
+
+The party host then provides the escrow transaction's `hash` value to the party band (receiver). The party band can use the `hash` value to look up the escrow transaction on the XRP Ledger to ensure that it was created according to the smart contract terms they agreed to.
```
Right. As part of this step, the band should probably confirm that the condition matches the one the oracle provided. If the condition's wrong, the fulfillment the oracle provides won't let the band get paid.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+Because participants want to create a conditionally-held escrow to provide the smart contract, they need a condition value and a fulfillment value. In this scenario, the participant that creates these values is the neutral party planner (oracle).
+
+The party planner generates the condition and fulfillment values and provides the condition value to the party host, who creates the escrow.
```
The party planner needs to tell the condition to the band also so that they know that this is the right condition.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+The party band (receiver) shows up and plays their set.
+
+The party planner (oracle) finishes the escrow using the fulfillment value they generated and kept secret.
```
Change this to reflect the band finishing the escrow using the fulfillment the oracle publishes.

mDuo13 replied:
Also note that this has to happen before 12 noon or the escrow expires and the band doesn't get paid.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+The party planner (oracle) finishes the escrow using the fulfillment value they generated and kept secret.
+
+Alternatively, if the party band is a no-show, the party planner does not finish the escrow. In this case, after 12 noon the next day, any participant can [cancel the escrow](cancel-an-expired-escrow.html) to return the held XRP to the party host's (sender's) account.
```
> the party planner does not finish the escrow

Change to:

> the party planner does not publish the fulfillment.

Also change "any participant can cancel..." to "anyone can cancel" because literally anyone can cancel an expired escrow.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+<span class="use-case-step-num">{{n.next()}}</span>
+## [Wait for validation](send-a-conditionally-held-escrow.html#7-wait-for-validation) and [confirm escrow finish](send-a-conditionally-held-escrow.html#8-confirm-final-result) (oracle, sender, and receiver)
+
+The party planner (oracle) waits for validation of the ledger that contains the escrow finish transaction and then confirms that the escrow was finished.
```
The band does this, but otherwise that's correct.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,149 @@
+
+The party planner (oracle) waits for validation of the ledger that contains the escrow finish transaction and then confirms that the escrow was finished.
+
+At this time, the party planner provides the transaction's `hash` value to the party host (sender) and party band (receiver). The party host and party band can use the `hash` value to look up the escrow transaction on the XRP Ledger to ensure that it is been finished correctly. The party host and party band can also just check their XRP Ledger account balances to ensure that their balances have d ...
```
The host's balance won't change at this step (unless the escrow was canceled) because the escrow creation already debited the locked-up XRP from their account.

mDuo13 reviewed (changes requested):
This use case isn't quite ready for primetime. There are some misconceptions about how fulfillments and expirations work, which I've tried to clarify. 

I also would really like to come up with a more plausible scenario than the party planning one, but I'm not sure I'll be able to come up with something. Still, I guess it's better to publish with this fanciful scenario than not to publish.

It might also be worth emphasizing that the oracle doesn't need an XRP Ledger account, or any secret communication channels... just a way to publish a condition and later a fulfillment reliably to the payer/payee, and a way to verify that the payee held up their end of the bargain.

jhaaaa committed:
escrow use case: rome's feedback

jhaaaa commented:
@mDuo13 - wow, wow, wow - thank you so much for such an informative and insightful review. I've addressed your feedback. Please let me know if there is anything else you'd like me to take a look at. 

If the content is good for now - I think we still need to address the use case template to be able to publish this. I think we discussed using some dactyl filters to allow the content to break some template rules? 😂

mDuo13 commented:
Yes, I looked into doing a Dactyl filter thing today, but at minimum it needs the bug fix that's in the forthcoming Dactyl version.

jhaaaa committed:
escrow use case: copyedits

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,135 @@
+
+- Should the escrow disallow fulfillment until a specific time?
+
+  While this is an option, the participants agree that it is unnecessary for their escrow. For conditionally-held escrows, enabling this option doesn't provide any additional security, since whether the escrow can be finished still depends entirely on whether the party planner (oracle) publishes the fulfillment before the expiration.
```
On this and the other "answers" in this Q&A-style section of this step, you need to indent the answers by 4 spaces instead of 2 so the answers are properly nested in the list items.

Also, you might consider bolding the "Questions" to make it a little easier to read.

jhaaaa committed:
escrow use case: template updates

jhaaaa commented:
👋 @warpaul and @mDuo13 - Thanks to Rome, we now have a Use Case template that can handle all kinds of formatting - so this escrow use case now looks presentable! 💅 

Any thoughts on our ability to publish this use case? I believe I've addressed all content feedback. However, I know that we agree that the scenario is a bit fanciful -- but do we think the content is accurate and valuable enough to publish? 

For the future, we can aim to come up with a more realistic use case that perhaps integrates with an actual software oracle?

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,137 @@
+
+- To be able to keep a secret (the fulfillment) until the time is right.
+
+- A way to communicate the fulfillment publicly or at least to the party band when the time is right.
```
One last thing the oracle must have:

- The ability to recognize whether the receiver (the party band) has fulfilled their end of the contract (played at the party).

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,137 @@
+
+      The participants agree that the escrow should potentially pay XRP to the party band's XRP Ledger account.
+
+
```
Another (arguably the most important) part of the puzzle: how shall the oracle know that the receiver did in fact play at the party?

If the party planner trusts the band and the party host to tell the truth, the oracle could hear conflicting reports about whether the band even played. Part of the job of an oracle is to know the truth (that's where the name comes from), so the party planner should be at the party to see for themselves.

You could picture a case where the party planner has some trusted partygoer(s) who report whether the band played, but in that case the partygoers and the party planner are jointly acting the role of oracle.

jhaaaa replied:
@mDuo13 -Yes, I see! In my mind, the party planner is at the party. Maybe I've seen too many movies with control-freak wedding planners in them so I just assumed the planner would be onsite. LOL. I will update the doc to make it clear that in this scenario, the party planner is at the party to witness the party band playing and therefore knows the truth firsthand.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,137 @@
+
+
+<span class="use-case-step-num">{{n.next()}}</span>
+## [Generate a condition and a fulfillment](send-a-conditionally-held-escrow.html#1-generate-condition-and-fulfillment) (oracle)
```
Similar to the Payment Channels case, I like to structure these sorts of headings as "Role: task" rather than "task (role)" because I prefer to set the context first.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,137 @@
+
+The party planner generates the condition and fulfillment values. The party planner provides the condition value to the party host, who creates the escrow. The part planner also provides the condition to the party band so that they know that this is the right condition.
+
+The party planner must keep the fulfillment value a secret. Anyone can use the condition and fulfillment values to finish the escrow, but in this scenario, it will most likely be the party band who will finish the escrow because they're the ones who are motivated to get paid.
```
> Anyone can use the condition and fulfillment values to finish the escrow, but in this scenario, it will most likely be the party band who will finish the escrow because they're the ones who are motivated to get paid.

I suggest breaking this up into two sentences and rewording to avoid future tense:

> Anyone can use the condition and fulfillment values to finish the escrow. Most often, the receiver finishes the escrow because they're the ones who are motivated to get paid.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,137 @@
+
+The party host (sender) waits for validation of the ledger that contains the escrow creation transaction and then confirms that the escrow was created.
+
+The party host then provides the escrow transaction's `hash` value to the party band (receiver). The party band can use the `hash` value to look up the escrow transaction on the XRP Ledger to ensure that it was created according to the smart contract terms they agreed to. As part of this step, the party band should also confirm that the condition matches the one the party planner (oracle) provide ...
```
> If the condition's wrong,

I suggest avoiding the contraction in this case and spelling out "condition is" because it's not a common contraction and it could be confused for a possessive by non-native speakers.

mDuo13 replied:
> As part of this step, the party band should also confirm that the condition matches

I suggest removing "also" from this phrase.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,137 @@
+
+The party planner (oracle) confirms that the party band has fulfilled their contract and publishes the fulfillment publicly, or at least to the party band.
+
+The party band finishes the escrow using the fulfillment value published by the party planner. The party band must finish the escrow before 12 noon or the escrow expires, the party band doesn't get paid, and the XRP returns to the party host's (sender's) XRP Ledger account.
```
> The party band must finish the escrow before 12 noon or the escrow expires, the party band doesn't get paid, and the XRP returns to the party host's (sender's) XRP Ledger account.

I suggest breaking this up into two sentences so that it doesn't read as a list. Also, the next paragraph explains how the XRP doesn't automatically return, so it's probably less confusing if you just leave that part off.

> The party band must finish the escrow before 12 noon. If they don't, the escrow expires and the party band doesn't get paid.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,137 @@
+
+The party band finishes the escrow using the fulfillment value published by the party planner. The party band must finish the escrow before 12 noon or the escrow expires, the party band doesn't get paid, and the XRP returns to the party host's (sender's) XRP Ledger account.
+
+Alternatively, if the party band is a no-show, the party planner does not publish the fulfillment. In this case, after 12 noon the next day, any participant can [cancel the escrow](cancel-an-expired-escrow.html) to return the held XRP to the party host's account.
```
I think you can rephrase this slightly to continue from either the "fulfillment not published" or "fulfillment published, but no one bothered to finish the escrow" cases:

> Alternatively, if the party band is a no-show, the party planner does not publish the fulfillment. In this case, the escrow cannot be finished. After 12 noon the next day, ...

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,137 @@
+At this time, the party band provides the transaction's `hash` value to the party host (sender). They can use the `hash` value to look up the escrow transaction on the XRP Ledger to ensure that it is been finished correctly.
+
+The party band can also just check their XRP Ledger account balance to ensure that their balance has increased by 2000 XRP. The party host's balance won't change at this step (unless the escrow was cancelled) because the escrow creation already debited the locked-up XRP from their account.
+<!-- USE_CASE_STEPS_START -->
```
This should be `<!-- USE_CASE_STEPS_END -->`

mDuo13 reviewed (changes requested):
Looking good. Left a few copy-edits and pointed out a couple small but important changes, but I think this is very close.

jhaaaa committed:
escrow use case: Rome's feedback

jhaaaa commented:
@mDuo13 - Thank you for yet another round of very thoughtful feedback. Please let me know if there is anything else you'd like to see implemented. 🙌

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,139 @@
+
+The party band must finish the escrow before 12 noon. If they don't, the escrow expires and the party band doesn't get paid.
+
+If the party planner does not publish the fulfillment (the party band is a no show) or if the party planner publishes the fulfillment, but no one finishes the escrow; after 12 noon the next day, any participant can [cancel the escrow](cancel-an-expired-escrow.html). Cancelling the escrow returns the held XRP to the party host's account.
```
Suggested change:

```
- any participant can
---
+ anyone can
```

because, _literally anyone_ can cancel an expired escrow.

mDuo13 commented on content/use-cases/use-an-escrow-as-a-smart-contract.md:
```diff
@@ -0,0 +1,139 @@
+
+At this time, the party band provides the transaction's `hash` value to the party host (sender). They can use the `hash` value to look up the escrow transaction on the XRP Ledger to ensure that it is been finished correctly.
+
+The party band can also just check their XRP Ledger account balance to ensure that their balance has increased by 2000 XRP. The party host's balance won't change at this step (unless the escrow was cancelled) because the escrow creation already debited the locked-up XRP from their account.
```
> The party band can also just check

Suggest simplifying this to:

> The party band can check

mDuo13 replied:
Also, _super-duper nit:_ prefer [the US spelling "canceled"](https://www.dictionary.com/browse/canceled) over the British spelling that has two L's.

mDuo13 reviewed (approved):
The revised version looks good. I left a couple copy-edits but I have no substantive improvements to suggest to the content.

jhaaaa committed:
esrow use case: rome's feedback
